File: Java/Okapi master/okapi service/src/com/tm/okapi/service/IOkapiService.java
package com.tm.okapi.service;

import java.util.AbstractMap;
import java.util.Date;
import java.util.HashMap;

import com.tm.okapi.service.OkapiService.*;

// NOTE: You can use the "Rename" command on the "Refactor" menu to change the interface name "IService1" in both code and config file together.
public interface IOkapiService
{
	byte[] createDocumentFromXliff(String sFileName, byte[] fileContent, String sFilterName, byte[] filterContent, String sourceLangISO639_1,
			String targetLangISO639_1, String sXliffContent) throws Exception;

	String createXliff(String sFileName, byte[] fileContent, String sFilterName, byte[] filterContent, String sourceLangISO639_1, String targetLangISO639_1,
			TMAssignment[] aTMAssignments) throws Exception;
}

File: .Net/CAT-service/BusinessServices/Okapi/IOkapiService.cs
using CAT.Models;

namespace CAT.BusinessServices.Okapi
{
    public interface IOkapiService
    {     
        string CreateXliffFromDocument(string fileName, byte[] fileContent, string filterName, byte[] filterContent, 
            string sourceLangISO639_1, string targetLangISO639_1);

        byte[] CreateDocumentFromXliff(string fileName, byte[] fileContent, string filterName, byte[] filterContent, 
            string sourceLangISO639_1, string targetLangISO639_1, string xliffContent);

        Statistics[] GetStatisticsForDocument(string sFileName, byte[] fileContent, string sFilterName, byte[] filterContent,
            string sourceLangISO639_1, string[] aTargetLangsISO639_1, TMAssignment[] aTMAssignments);

        string PreTranslateXliff(string sXliffContent, string langFrom_ISO639_1, string langTo_ISO639_1, 
            TMAssignment[] aTMAssignments, int matchThreshold);
    }
}